packages/@tailwindcss-upgrade/src/codemods/migrate-config.ts

```diff
@@ -63,7 +63,22 @@ export function migrateConfig(
           plugin.path[0] === '.'
             ? relativeToStylesheet(sheet, path.resolve(plugin.base, plugin.path))
             : plugin.path
-        css += `@plugin '${relative}';\n`
+
+        if (plugin.options === null) {
+          css += `@plugin '${relative}';\n`
+        } else {
+          css += `@plugin '${relative}' {\n`
+          for (let [property, value] of Object.entries(plugin.options)) {
+            css += `  ${property}: ${
+              Array.isArray(value)
+                ? value.map((v) => (typeof v === 'string' ? quoteString(v) : '' + v)).join(', ')
+                : typeof value === 'string'
+                  ? quoteString(value)
+                  : '' + value
+            };\n`
```

Comment:
I think I'd split this into multiple concatenations w/ ifs wrapping things rather than inlining all of them. It's feels a bit hard to follow like this.